### PR TITLE
[4.0] Save sidebar state using localStorage

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #15892 

### Summary of Changes

This PR adds the use of `localStorage` to save the state of the sidebar (open/closed). Upon page refresh it wil automatically open or close the sidebar based on the state it was in before.

### Testing Instructions

**1.**
- Apply patch (sidebar should be open)

**2.** 
- Close the sidebar
- Refresh the page
- Sidebar should stay closed

**3.** 
- Open the sidebar
- Refresh the page
- Sidebar should stay open

@dgt41 I've added the test to see if `localStorage` can be used in the `core.js`, so you can use this from now on. Just do `if (Joomla.localStorageEnabled()) { ... }`

### CODE REVIEW

**Comment by dgt41** [comment]
@C-Lodder do we need this check?
Looking at https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage it should be available for all supported browsers in J4

**Comment by C-Lodder** [comment]
Yup, but if the user has it disabled on their browser for some reason, then this is just a sanity check to prevent any console errors, which may then stop something else from working.

**Comment by brianteeman** [comment]
brings up a good point though - do we have a list of supported MOBILE browsers

**Comment by dgt41** [comment]
@brianteeman no AFAIK

**Comment by C-Lodder** [comment]
@brianteeman - You're not going to want to keep the sidebar open on mobile

File: administrator/templates/atum/js/template.js [hunk 1]
```diff
@@ -28,22 +29,33 @@
 			return null;
 		}
 
+		var wrapper = document.getElementById('wrapper'),
+		    sidebar = document.getElementById('sidebar-wrapper');
+
+		// Set the initial state of the sidebar based on the localStorage value
+		if (Joomla.localStorageEnabled()) {
```
Comment (dgt41) [inline administrator/templates/atum/js/template.js]:
```js
if (window.localStorage !== null)
```
same functionality

**Comment by brianteeman** [comment]
@c-lodder it was an offtopic comment that we should probably have a list of supported mobile browsers. And don't forget some "mobile" browsers run on very large devices now. Mobile browser doesn't equal small screen.

File: administrator/templates/atum/js/template.js (hunk 1 above)
Comment (dgt41) [inline administrator/templates/atum/js/template.js]: it won't work for safari

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 8c94b9dd22b314dcffd12d0dd85a1cc285c5fa88<br /><br />With this PR certain views end up with an empty space at the side 

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16036">issues.joomla.org/tracker/joomla-cms/16036</a>.</sub>

**Comment by C-Lodder** [comment]
which views? edit?

**Comment by brianteeman** [comment]
## before pr
screenshotr17-27-35

## after pr
screenshotr17-28-19

**Comment by brianteeman** [comment]
@C-Lodder you were too fast i was uploading the images ;)

**Comment by C-Lodder** [comment]
Will fix tomorrow morning

**Comment by C-Lodder** [comment]
@brianteeman - Fixed the edit views. Would you mind retesting please?

**Pull request merged**